packages/grafana-prometheus/src/querybuilder/components/metrics-modal/helpers.ts

```diff
@@ -0,0 +1,144 @@
+// Core Grafana history https://github.com/grafana/grafana/blob/v11.0.0-preview/public/app/plugins/datasource/prometheus/querybuilder/components/metrics-modal/state/helpers.ts
+import { t } from '@grafana/i18n';
+
+import { PrometheusLanguageProviderInterface } from '../../../language_provider';
+
+import { MetricData, MetricsData, PromFilterOption } from './types';
+
+// Constants
+const HISTOGRAM_TYPES = ['histogram', 'summary'] as const;
+const OLD_HISTOGRAM_PATTERN = /^\w+_bucket$|^\w+_bucket{.*}$/;
+const HISTOGRAM_TYPE = 'histogram';
+const NATIVE_HISTOGRAM_TYPE = 'native histogram';
+
+/**
+ * Builds the metric data object with type and description
+ * @param metric - The metric name
+ * @param languageProvider - The Prometheus language provider interface
+ * @returns MetricData object with value, type, and description
+ */
+export const generateMetricData = (
+  metric: string,
+  languageProvider: PrometheusLanguageProviderInterface
+): MetricData => {
+  const metadata = languageProvider.retrieveMetricsMetadata();
+
+  let type = metadata[metric]?.type;
+  const description = metadata[metric]?.help;
+
+  HISTOGRAM_TYPES.forEach((t) => {
+    if (description?.toLowerCase().includes(t) && type !== t) {
+      type = type ? `${type} (${t})` : t;
+    }
+  });
+
+  const oldHistogramMatch = metric.match(OLD_HISTOGRAM_PATTERN);
+
+  if (type === HISTOGRAM_TYPE && !oldHistogramMatch) {
+    type = NATIVE_HISTOGRAM_TYPE;
+  }
+
+  return {
+    value: metric,
+    type: type,
+    description: description,
+  };
+};
+
+export function calculatePageList(metricsData: MetricsData, resultsPerPage: number): number[] {
+  if (!Array.isArray(metricsData) || metricsData.length === 0) {
+    return [];
+  }
+
+  if (resultsPerPage <= 0) {
+    return [1];
+  }
+
+  const totalPages = Math.ceil(metricsData.length / resultsPerPage);
+  return Array.from({ length: totalPages }, (_, i) => i + 1);
+}
+
+export const calculateResultsPerPage = (results: number, defaultResults: number, max: number): number => {
+  if (!Number.isInteger(results) || results < 1) {
+    return Math.max(1, defaultResults);
+  }
+
+  if (results > max) {
+    return max;
+  }
+
+  return results;
+};
+
+export const getPromTypes: () => PromFilterOption[] = () => [
```

Comment:
These are not new. I moved helpers.ts file to the `metrics-modal` folder from `state` folder.